### PR TITLE
Update IntelliJ and Kotlin Gradle plugins

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -14,8 +14,8 @@ import java.util.zip.ZipEntry
 import java.util.zip.ZipOutputStream
 
 plugins {
-    kotlin("jvm") version "1.5.31"
-    id("org.jetbrains.intellij") version "1.2.1"
+    kotlin("jvm") version "1.7.0"
+    id("org.jetbrains.intellij") version "1.7.0-SNAPSHOT"
```
Comment (henryju) [inline build.gradle.kts:18]: Normally the final 1.7.0 release should be done this week.

File: its/build.gradle.kts [hunk 2]
```diff
@@ -20,9 +20,9 @@ val compileTestKotlin: org.jetbrains.kotlin.gradle.tasks.KotlinCompile by tasks
 compileTestKotlin.kotlinOptions.jvmTarget = "11"
 
 repositories {
-    mavenCentral()
     maven("https://repox.jfrog.io/repox/sonarsource")
     maven("https://packages.jetbrains.team/maven/p/ij/intellij-dependencies")
+    mavenCentral()
```
Comment (damien-urruty-sonarsource) [inline its/build.gradle.kts:25]: :+1: good move for security!

**Review by damien-urruty-sonarsource — APPROVED** [review]

File: settings.gradle.kts [hunk 3]
```diff
@@ -1,3 +1,9 @@
 rootProject.name = "sonarlint-intellij"
 include("its", "clion", "common", "git", "rider")
 
+pluginManagement {
+    repositories {
+        maven("https://oss.sonatype.org/content/repositories/snapshots/")
+        gradlePluginPortal()
```
[the remark this answers is not in the document]
Comment (henryju) [reply settings.gradle.kts:7]: Yes

File: its/build.gradle.kts [hunk 4]
```diff
@@ -35,8 +35,9 @@ dependencies {
     testImplementation("org.sonarsource.sonarqube:sonar-ws:8.5.1.38104")
     testImplementation("com.intellij.remoterobot:remote-robot:$remoteRobotVersion")
     testImplementation("com.intellij.remoterobot:remote-fixtures:$remoteRobotVersion")
-    testImplementation("org.junit.jupiter:junit-jupiter-api:5.7.0")
-    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.7.0")
+    testImplementation("org.junit.jupiter:junit-jupiter-api:5.8.2")
+    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.8.2")
```
[the remark this answers is not in the document]
Comment (henryju) [reply its/build.gradle.kts:39]: No idea what is best. On the JUnit project, they also use the BOM: https://github.com/junit-team/junit5-samples/blob/5f9e577e2c64b06f8b7ef5633462144feed97fef/junit5-jupiter-starter-gradle/build.gradle#L12

[the remark this answers is not in the document]
Comment (henryju) [reply its/build.gradle.kts]: I went for official JUnit suggestion

**Pull request merged**